### PR TITLE
MinRssBrakingDistance for all Vehicles and Fixed ComputeSafetyPolygon

### PR DESCRIPTION
**Pull request by patrickhart**
* Fixes segfault issue with `ComputeSafetyPolygon` when the other agent has already been deleted
* Implements a calculation function for the minimum braking distance of an agent `ComputeMinBrakingPolygon`
* Modified viewer that draws all the braking distances using the flag `params["Visualization"]["Evaluation"]["DrawMinRSSBrakingDistances"] = True`

### CODE REVIEW

File: bark/runtime/viewer/viewer.py [hunk 1]
```diff
@@ -481,9 +489,26 @@ def drawRssSafetyResponses(self, world, ego_id):
             self.drawPolygon2d(transformed_polygon, response_color,
                                0.6, response_color, zorder=9)
 
+    def DrawRSSBrakingDistances(self, world, eval_id):
```
Comment (klemense1) [inline bark/runtime/viewer/viewer.py:492]: does this also work when using the EvaluatorRSS and not the BehaviorRSSCompliant? Maybe add a docstring to indicate usage?

File: bark/world/evaluation/rss/safety_polygon.hpp [hunk 2]
```diff
@@ -53,6 +53,18 @@ struct SafetyPolygon {
 
 typedef std::shared_ptr<SafetyPolygon> SafetyPolygonPtr;
 
+/**
+ * @brief  Modes for drawing the longitudinal visualization
+ * @note   
+ * @retval None
+ */
+enum LonDirectionMode {
+  AUTO = 0,
```
Comment (klemense1) [inline bark/world/evaluation/rss/safety_polygon.hpp:62]: what does auto mean?

File: bark/world/evaluation/rss/safety_polygon.hpp [hunk 3]
```diff
@@ -102,24 +118,39 @@ inline void ComputeSafetyPolygon(
...
+    if (directional == LonDirectionMode::AUTO) {
+      // calculate if the agent is in front or behind
+      auto other_agent = observed_world.GetAgents()[safe_poly.agent_id];
+      if (!other_agent)
+        return;
+      auto other_pose = other_agent->GetCurrentPosition();
+      double relative_angle = atan2(
+        bg::get<1>(ego_pose) - bg::get<1>(other_pose),
+        bg::get<0>(ego_pose) - bg::get<0>(other_pose));
+      double diff_angle = SignedAngleDiff(theta - M_PI_2, relative_angle);
```
Comment (klemense1) [inline bark/world/evaluation/rss/safety_polygon.hpp:131]: why M_PI_2? is it then always within [-pi, pi]?

**Review by klemense1 — CHANGES_REQUESTED** [review]
Nice! have a look on my comment towards the valid range of theta

File: bark/world/evaluation/rss/safety_polygon.hpp (hunk 3 above)
Comment (patrickhart) [reply bark/world/evaluation/rss/safety_polygon.hpp:131]: this is a 90 degree turn

File: bark/world/evaluation/rss/safety_polygon.hpp (hunk 2 above)
Comment (patrickhart) [reply bark/world/evaluation/rss/safety_polygon.hpp:62]: it either draws the polygon in front or behind based on the other agent. it is stated in the function itself

File: bark/runtime/viewer/viewer.py (hunk 1 above)
Comment (patrickhart) [reply bark/runtime/viewer/viewer.py:492]: Yes, it always works. Thus, it is nor required in my opinion

File: bark/world/evaluation/rss/commons.hpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  Acceleration brake_min_correct_lon = Acceleration(rss_params->GetReal(
+    "BrakeLonMinCorrect", "minimum deceleration of oncoming vehicle", -1.67));
+  Duration response_time = Duration(
+    rss_params->GetReal("TimeResponse", "response time of the ego vehicle", 0.2));
+  Distance min_stopping_distance_lat;
+  Acceleration accel_max_lat = Acceleration(
+    rss_params->GetReal("AccLatBrakeMax", "maximum lateral acceleration", 0.2));
+  Acceleration brake_min_correct_lat = Acceleration(
+    rss_params->GetReal("AccLatBrakeMin", "minimum lateral braking", -0.8));
+
```
Comment (tobiaskessler) [inline bark/world/evaluation/rss/commons.hpp:69]: are these the default values from Intel?

File: bark/world/evaluation/rss/safety_polygon.hpp [hunk 5]
```diff
@@ -65,9 +77,13 @@ typedef std::shared_ptr<SafetyPolygon> SafetyPolygonPtr;
  * @retval None
  */
 inline void ComputeSafetyPolygon(
```
Comment (tobiaskessler) [inline bark/world/evaluation/rss/safety_polygon.hpp:79]: return bool if success or not?

**Review by tobiaskessler — APPROVED** [review]
looks good, only minor stuff, see my comments

File: bark/world/evaluation/rss/commons.hpp (hunk 4 above)
Comment (patrickhart) [reply bark/world/evaluation/rss/commons.hpp:69]: Yes, these are the same parameters as in the BehaviorRSSConformant.
Ideally, these should be specified in a parameter file, but if not, these should still make sense.

**Review by klemense1 — APPROVED** [review]

**Pull request merged**